Frontend/ui-customer/src/Screens/CustomerUtils/ReportIssueChatScreen/Controller.purs

```diff
@@ -0,0 +1,475 @@
+  let messages' = snoc state.data.chatConfig.messages (ChatView.makeChatComponent optionName "Customer" (getCurrentUTC ""))
+  case find (\x -> x.option == optionName) state.data.options of
+    Just selectedOption -> do
+      if (selectedOption.label == "CALL_DRIVER")then continue state { props { showCallDriverModel = true, isPopupModelOpen = true }, data{ selectedOption = Just selectedOption }} 
+        else  if (selectedOption.label == "REOPEN_TICKET") then exit $ ReopenIssue state { data{ selectedOption = Just selectedOption ,  options = [], chatConfig {suggestionsList = [], messages = messages'}} }
+          else do
+            pure $ if (selectedOption.label == "DOWNLOAD_INVOICE") then downloadInvoicePDF state else unit
+            exit (SelectIssueOption (state { data { chatConfig { messages = messages', suggestionsList = []  }, selectedOption = Just selectedOption } }))
+    Nothing -> do
+      _ <- pure $ toast "Can't find option"
+      continue state
+
+eval (ChatViewActionController (ChatView.BackPressed)) state = do
+  _ <- pure $ hideKeyboardOnNavigation true
+  continueWithCmd state [do
+    pure $ BackPressed
+  ]
+
+eval (ChatViewActionController (ChatView.OnImageClick url)) state = do
+  continue state { data  { viewImageState { image = url, imageName = Nothing } }
+                 , props { showViewImageModel = true } }
+
+eval (ChatViewActionController (ChatView.EnableSuggestions)) state =
+  continue state { data { chatConfig { enableSuggestionClick = true } } }
+
+eval (SendMessage message toggleSubmitComp) state = do
+  let messages'       = snoc state.data.chatConfig.messages message
+  let showSubmitComp' = if toggleSubmitComp then not state.props.showSubmitComp else state.props.showSubmitComp
+  continueWithCmd state { data  { chatConfig { messages = messages' } }
+                        , props { showSubmitComp = showSubmitComp' } } [do
+    _ <- scrollToEnd (getNewIDWithTag "ChatScrollView") true
+    pure NoAction
+  ]
+
+eval (UpdateState updatedState) state = do
+  continue updatedState
+
+eval (PrimaryEditTextActionController (PrimaryEditText.TextChanged _ text)) state =
+  continue state { data { messageToBeSent = if (ST.length text >= 256) then state.data.messageToBeSent else text } }
+
+eval (ConfirmCall (PrimaryButton.OnClick)) state =
+  case find (\x -> (state.props.showCallSupportModel && x.label == "CALL_SUPPORT") ||(state.props.showCallDriverModel && x.label == "CALL_DRIVER")) state.data.options of
+    Just selectedOption -> do
+      let messages' = snoc state.data.chatConfig.messages (ChatView.makeChatComponent selectedOption.option "Customer" (getCurrentUTC "") )
+      if state.props.showCallDriverModel
+      then
+        exit $ CallDriver state { data { chatConfig { messages = messages', suggestionsList = [] }, selectedOption = Just selectedOption }, props { isPopupModelOpen = false, showCallDriverModel = false, showCallSupportModel = false } }
```

Comment:
store the state in a variable and the use it while exiting the screen , as it is being repeated twice.